doc/design.md

```diff
@@ -442,3 +500,356 @@ Covers:
 * [`req~npm-changed-dependency~1`](system_requirements.md#get-changed-dependency)
 
 Needs: impl, utest, itest
+
+## Automatic Dependency Update Process
+
+This consists of the following steps:
+1. Trigger the dependency update process
+2. Update dependencies
+3. Create a pull request
+
+![Activity Diagram for the dependencies update process](images/dependencies_update_process.svg)
+
+#### Triggering the Dependency Update Process
+`dsn~trigger-dependency-updates~1`
+
+PK generates the `dependencies_check.yml` GitHub workflow so that it launches the `dependencies_update.yml` workflow when it detects new vulnerabilities.
+
+Rationale:
+
+`dependencies_check.yml` already uses the [security-issues](https://exasol.github.io/python-toolbox/github_actions/security_issues.html) tool from the [python-toolbox](https://github.com/exasol/python-toolbox) to create issues for new vulnerabilities. Re-implementing this in PK is not necessary.
+
+Covers:
+* [`req~auto-update-dependencies~1`](system_requirements.md#auto-update-dependencies)
+
+-Needs: impl, utest, itest
+
+#### Update Dependencies Mode
+`dsn~update-dependencies-mode~1`
+
+PK provides an `update-dependencies` mode in addition to `fix` and `verify`. This mode performs the following steps:
+
+1. Increment version of the project
+2. Update dependencies to their latest version
+3. Create changelog containing information about the fixed vulnerabilities (if available)
+
+Rationale:
+
+* We implement this in PK because
+  * PK already contains code for working with versions and changelog, so we can reuse this code
+  * The `update-dependencies` mode is also useful for running locally on the developer's machine when working on a non-security related task
+* We don't implement git/GitHub operations in PK because
+  * This would couple PK to GitHub
+  * This would be surprising when running it locally
+  * This would require credentials for accessing the GitHub API
```

Comment:
ah ok and this answers my question from the diagrams, but maybe think about two modes, one which runs the Git/Github things and another doesn't; otherwise you probably will create scripts for creating commit and PR which is usually a maintenance hell.

Reply:
If the workflow is not maintainable I will move this to PK.